### PR TITLE
Tdr 469 ecr scan notifications

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

**Review by TomJKing — DISMISSED** [review]

File: modules/permissions/common_permissions.tf [hunk 1]
```diff
@@ -122,4 +122,20 @@ resource "aws_iam_policy" "custodian_get_parameters" {
   name        = "TDRCustodianGetParameters"
   description = "Policy to allow Cloud Custodian to get SSM parameters"
   policy      = data.aws_iam_policy_document.custodian_get_parameters.json
-}
\ No newline at end of file
+}
+
```
Comment (TomJKing) [inline modules/permissions/common_permissions.tf:126]: Extra line break

File: modules/permissions/common_permissions.tf [hunk 2]
```diff
@@ -122,4 +122,20 @@ resource "aws_iam_policy" "custodian_get_parameters" {
...
+
+resource "aws_iam_role" "jenkins_lambda_deploy_role" {
+  name = "TDRJenkinsNodeLambdaRoleMgmt"
```
Comment (TomJKing) [inline modules/permissions/common_permissions.tf:129]: Should environment "Mgmt" be hardcoded?

File: modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      "Sid": "",
+      "Effect": "Allow",
+      "Action": [
+        "lambda:UpdateFunctionCode",
+        "lambda:PublishVersion",
+        "s3:PutObject",
+        "s3:GetObject"
+      ],
+      "Resource": [
+        "arn:aws:lambda:eu-west-2:${account_id}:function:tdr-ecr-scan-notifications-mgmt",
```
Comment (TomJKing) [inline modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl:14]: Should environment "mgmt" be hardcoded?

File: modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        "arn:aws:s3:::tdr-backend-code-mgmt/*"
+      ]
+    }
+  ]
+}
```
Comment (TomJKing) [inline modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl:19]: Missing line break

File: modules/permissions/variables.tf [hunk 5]
```diff
@@ -31,4 +31,6 @@ variable "release_bucket_arn" {
 variable "staging_bucket_arn" {
   description = "The bucket arn for staging artefacts"
   type        = string
-}
\ No newline at end of file
+}
+
+variable "management_account_number" {}
```
Comment (TomJKing) [inline modules/permissions/variables.tf:36]: Missing line break

File: modules/permissions/common_permissions.tf [hunk 6]
```diff
@@ -122,4 +122,20 @@ resource "aws_iam_policy" "custodian_get_parameters" {
   name        = "TDRCustodianGetParameters"
```
Comment (TomJKing) [inline modules/permissions/common_permissions.tf:122]: Has `terraform fmt` been run on this file?

File: root.tf [hunk 7]
```diff
@@ -155,6 +155,7 @@ module "common_permissions" {
   release_bucket_arn             = module.release_artefacts_s3.s3_bucket_arn
   staging_bucket_arn             = module.staging_artefacts_s3.s3_bucket_arn
   terraform_scripts_state_bucket = module.terraform_state.terraform_scripts_state_bucket_arn
+  management_account_number = data.aws_ssm_parameter.mgmt_account_number.value
```
Comment (TomJKing) [inline root.tf:158]: Has `terraform fmt` been run on this file?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a few minor comments

File: modules/permissions/common_permissions.tf (hunk 6 above)
Comment (MancunianSam) [reply modules/permissions/common_permissions.tf:122]: No, I've run it now and on the other files.

File: modules/permissions/common_permissions.tf (hunk 1 above)
Comment (MancunianSam) [reply modules/permissions/common_permissions.tf:126]: I've removed this one.

File: modules/permissions/common_permissions.tf (hunk 2 above)
Comment (MancunianSam) [reply modules/permissions/common_permissions.tf]: Technically it's fine but it is a bit rubbish. I've added a variable in and passed it in hard coded from the main root.tf file like the other modules.

File: modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl (hunk 3 above)
Comment (MancunianSam) [reply modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl:14]: Same with this one and the s3 bucket in the policy, I've used the variable.

File: modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl (hunk 4 above)
Comment (MancunianSam) [reply modules/permissions/templates/jenkins_lambda_deploy_policy.json.tpl:19]: I've added this one and the one in the variables file.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**